libs/go/libp2p_node/aea/api.go

```diff
@@ -397,23 +397,84 @@ func (aea *AeaApi) stop() {
   Pipes helpers
 
 */
+const CurrentVersion = "0.1.0"
 
-func write_envelope(pipe Pipe, envelope *Envelope) error {
-	data, err := proto.Marshal(envelope)
+func MakeACNMessageFromEnvelope(envelope *Envelope) (error, []byte) {
+	envelope_bytes, err := proto.Marshal(envelope)
+	if err != nil {
+		return err, envelope_bytes
+	}
+	return acn.EncodeACNEnvelope(envelope_bytes)
+}
+
+func (aea AeaApi) SendEnvelope(envelope *Envelope) error {
+	err, data := MakeACNMessageFromEnvelope(envelope)
 	if err != nil {
 		logger.Error().Str("err", err.Error()).Msgf("while serializing envelope: %s", envelope)
 		return err
 	}
-	return pipe.Write(data)
+	err = aea.pipe.Write(data)
+	if err != nil {
+		logger.Error().Str("err", err.Error()).Msgf("on pipe write")
+		return err
+	}
+
+	status, err := acn.WaitForStatus(aea.acn_status_chan, ACN_STATUS_TIMEOUT)
```

Comment:
Does this work well under concurrency?

Reply:
read is a go routine, write is not, and write is blocked till confirmation arrived
probably need to introduce write loop go routine  like it's done in python

Reply:
done